lib/ui/screens/post_create_screen/blocs/post_input/post_input_state.dart

```diff
@@ -75,7 +75,7 @@ class PostInputState extends Equatable {
       allowsComments: allowsComments ?? this.allowsComments,
       saving: saving ?? this.saving,
       medias: medias ?? this.medias,
-      poll: poll ?? this.poll,
+      poll: poll == null ? poll : poll ?? this.poll,
```

Comment:
What's happening here? Isn't `poll == null ? poll : poll` the same as simply writing `poll` ?

Reply:
Probably instead of doing this strange implementation, we could simply add a method to the state as 

```dart
PostInputState removePoll() {
  return PostInputState(
    // copy all the values
    poll: null
  );
```

Reply:
Oh there was a brainfart moment. But i think for the copyWith function we should allow null types and only default values on undefined

```
PostInputState copyWith({
    String message,
    bool allowsComments,
    List<PostMedia> medias,
    PostPoll poll,
    bool showPopup,
    bool saving,
    bool willShowPopupAgain,
  }) {
    return PostInputState(
      parentPost: parentPost,
      message: message != undefined ? message : this.message,
      allowsComments: allowsMessage != undefined ? allowsComments : this.allowsComments,
     ...etc
    );
  }
```

Reply:
@ryuash Dart has no such thing as undefined unluckily. It only has null as the default value for objects that are not initialized

Reply:
@RiccardoM oh would this work then? Instead of checking we have the states in the default value that way I should be able to pass an intentional null.
```
PostInputState copyWith({
    String message = this.message,
    bool allowsComments = this.allowsComments,
    List<PostMedia> medias = this.medias,
    ...etc
  }) {
    return PostInputState(
      parentPost: parentPost,
      message: message,
      allowsComments: allowsComments,
     ...etc
    );
  }
```

I updated the PR btw should be good to go after another review

Reply:
@ryuash That would work perfectly!